api/proto/config/acipherinfo.proto

```diff
@@ -64,10 +64,13 @@ message CipherBlock {
 
 
 // This message will be filled with the
-// credential details and encrypted across
-//  wire for data in transit, by the controller
+// details to be encrypted and shared across
+// wire for data in transit, by the controller
 // for encryption
```

Comment:
I think you need to comment that different subsets of these fields are filled in for different usage of this block.
Doesn't this mean that you'll end up with different code to handle the decrypt? Or the different use cases pick up different fields.

Reply:
Hi @eriknordmark,  the decrypt logic will be common, and give back the structure, and individual agents will pick up different fields post decryption.